### PR TITLE
proxy: disable skipping DNS lookup by default and add a knob to control it

### PR DESCRIPTION
**Pull request by Augustyniak**
Description: Add an engine builder knob for controlling whether an upstream change for skipping a wait on a DNS resolution for proxied request (https://github.com/envoyproxy/envoy/pull/23294) is enabled or not. Disable the feature by default.
Risk Level: Low, adds feature flag that disable a risky change by default and allows for controlling as to whether it's enabled or not.
Testing: Example apps
Docs Changes: N/A
Release Notes: Done

Signed-off-by: Rafal Augustyniak <raugustyniak@lyft.com>

### CODE REVIEW

File: library/java/io/envoyproxy/envoymobile/engine/EnvoyConfiguration.java [hunk 1]
```diff
@@ -89,21 +90,26 @@ public enum TrustChainVerification {
...
-   * @param httpPlatformFilterFactories  the configuration for platform filters.
-   * @param stringAccessors              platform string accessors to register.
-   * @param keyValueStores               platform key-value store implementations.
+   * @param h2ConnectionKeepaliveTimeoutSeconds           rate in seconds to timeout h2 pings.
+   * @param h2ExtendKeepaliveTimeout                      Extend the keepalive timeout when *any*
+   *     frame is received
+   *                                                      on the owning HTTP/2 connection.
+   * @param maxConnectionsPerHost                         maximum number of connections to open to a
+   *     single host.
+   * @param statsFlushSeconds                      interval at which to flush Envoy stats.
```
Comment (jpsim) [inline library/java/io/envoyproxy/envoymobile/engine/EnvoyConfiguration.java:99]: Nit: the indentation of the comment bodies here appear inconsistent

**Review by jpsim — DISMISSED** [review]

**Review by jpsim — DISMISSED** [review]

**Comment by Augustyniak** [comment]
@jpsim needed to push another change to fix cronet engine builder. Mind re-approving?

**Review by jpsim — DISMISSED** [review]

**Review by jpsim — APPROVED** [review]

**Review by jpsim — APPROVED** [review]

**Pull request merged**